packages/vscode-extension/src/debug/prerequisitesHandler.ts

```diff
@@ -104,6 +139,67 @@ async function checkM365Account(): Promise<CheckFailure | undefined> {
   }
 }
 
+async function checkDependencies(
+  localEnvManager: LocalEnvManager,
+  depsChecker: VSCodeDepsChecker,
+  projectSettings: ProjectSettings
+): Promise<CheckFailure | undefined> {
+  try {
+    const deps = localEnvManager.getActiveDependencies(projectSettings);
+
+    const resolveRes = await depsChecker.resolve(deps);
+    if (resolveRes.isErr()) {
+      return {
+        checker: "Dependencies",
+        error: returnUserError(
+          resolveRes.error,
+          ExtensionSource,
+          ExtensionErrors.PrerequisitesValidationError,
+          resolveRes.error.helpLink
+        ),
+      };
+    }
+
+    return undefined;
+  } catch (error: any) {
+    return {
+      checker: "Dependencies",
+      error: assembleError(error),
+    };
+  }
+}
+async function resolveBackendExtension(
+  depsChecker: VSCodeDepsChecker,
+  workspacePath: string,
+  projectSettings: ProjectSettings
+): Promise<CheckFailure | undefined> {
+  try {
+    if (ProjectSettingsHelper.includeBackend(projectSettings)) {
+      const backendRoot = path.join(workspacePath, FolderName.Function);
+
+      const res = await installBackendExtension(backendRoot, depsChecker, vscodeLogger);
```

Comment:
Does `installBackendExtension` always install or check-then-install?

Reply:
Now it always install. There is no check logic in deps checker.

Reply:
How long for the second run? Add `TODO` to backend extension install for further optimization.

Reply:
It always runs `dotnet build...`, but uses dotnet's cache on the second run.